### DISCUSSION TITLE
[skywalking-go]When I was using the Skywalk-go agent, I had a problem with not being able to build the project

### DISCUSSION THREAD

**Discussion by pinkfloyd-eminem** [apache/skywalking#11188] [Q&A] [answered]
U-VF4VY9W5-1944:skywalking-go-test whlongxi$ sudo go build -toolexec="./go-agent" -a
Password:
go: error obtaining buildID for go tool compile: fork/exec ./go-agent: permission denied

I referred to https://skywalking.apache.org/docs/skywalking-go/v0.2.0/en/setup/gobuild/。

how to solve this problem？

**Comment by wu-sheng** [comment]
This seems a basic Linux issue. Do you add sudo or grant execute to the file?

**Reply by pinkfloyd-eminem** [reply]
i use sudo and set chmod 777

**Reply by pinkfloyd-eminem** [reply]
and i test both macos and ubuntu18.04

**Reply by wu-sheng** [reply]
But this error is a OS-level thing. @mrproliu Do you expect anything else?

**Reply by mrproliu** [reply]
I think it's related to the execute working directory, have you tried the absolute path of the agent?

**Reply by pinkfloyd-eminem** [reply]
yes，I started with the absolute path of the agent, and then copied the agent into the project and used the relative path, which didn't work

**Reply by pinkfloyd-eminem** [reply]
Neither the absolute path nor the relative path.Is there any official image? The image contains the example of skywalk-go accessing demo

**Reply by mrproliu** [reply]
> Is there any official image? 

[Here is an example ](https://github.com/apache/skywalking-showcase/tree/main/services/rating-service) using the skywalking-go docker image. 

Also, have you checked your private access control on MacOS? they maybe need you to trust the download execute the file.

**Reply by pinkfloyd-eminem** [reply]
But I also tried in another Alibaba Cloud ecs environment and encountered the same problem.

**Reply by mrproliu** [reply]
I have tried it in the cloud and it works well. I'm not sure what's happened to you. 
Have you tried to execute the agent program directly? Such as `/path/to/skywalking-go-agent-0.2.0-linux-amd64 --help`?
If you are unable to work properly, please make sure that the downloaded binary file is correct, the platform is correct, or if there are relevant permissions.

**Comment by pinkfloyd-eminem** [comment, marked as answer]
Thank you. I have successfully connected to Skywalk-go agent. Thank you for your help. My steps are as follows:
1.cd skywalking-go/
2.go build
3.sudo go build -toolexec "/path/to/skywalking-go/bin/skywalking-go-agent--darwin-amd64 -config /path/to/my_agent_config.yaml" -a   （The path of the executable file is an absolute path）

**Reply by mrproliu** [reply]
It's another way to build the agent from the source, you could follow [this documentation if needed](https://skywalking.apache.org/docs/skywalking-go/next/en/development-and-contribution/build-and-use-agent/).

**Comment by pinkfloyd-eminem** [comment]
Thank you. I have successfully connected to Skywalk-go agent. Thank you for your help. My steps are as follows:
1.cd skywalking-go/
2.go build
3.sudo go build -toolexec "/path/to/skywalking-go/bin/skywalking-go-agent--darwin-amd64 -config /path/to/my_agent_config.yaml" -a   （The path of the executable file is an absolute path）
